erts/emulator/sys/common/erl_check_io.c

```diff
@@ -798,7 +801,7 @@ driver_select(ErlDrvPort ix, ErlDrvEvent e, int mode, int on)
     void (*stop_select_fn)(ErlDrvEvent, void*) = NULL;
     Port *prt = erts_drvport2port(ix);
     Eterm id = erts_drvport2id(ix);
-    ErtsSysFdType fd = (ErtsSysFdType) e;
+    ErtsSysFdType fd = (ErtsSysFdType) event2fd(e);
```

Comment:
This all looks good for Unix where `ErtsSysFdType` is `int`, but on **Windows** it's defined as `HANDLE` which in turn is a pointer I think.

Reply:
Ah, good point!